tests/test_jwt_utils.py

```diff
@@ -0,0 +1,92 @@
+from datetime import datetime, timedelta
+from unittest.mock import patch, MagicMock
+
+import pytest
+from jose import jwt
+
+from lbz.authz.authorizer import Authorizer
+from lbz.exceptions import Unauthorized
+from lbz.jwt_utils import get_matching_jwk, decode_jwt
+from tests import SAMPLE_PUBLIC_KEY, SAMPLE_PRIVATE_KEY
+
+BAD_PUBLIC_KEY = {
+    "kid": "wrong-key",
+}
+
+
+class TestGetMatchingJWK:
+    @patch.object(jwt, "get_unverified_header", return_value=SAMPLE_PUBLIC_KEY)
+    def test_get_matching_key(self, get_unverified_header_mock: MagicMock) -> None:
+        assert get_matching_jwk("x") == SAMPLE_PUBLIC_KEY
+        get_unverified_header_mock.assert_called_once()
+
+    @patch.object(jwt, "get_unverified_header", return_value=BAD_PUBLIC_KEY)
+    def test_get_matching_key_fail(self, _get_unverified_header_mock: MagicMock) -> None:
+        with pytest.raises(Unauthorized):
+            get_matching_jwk("x")
+
+    @patch.object(jwt, "get_unverified_header", return_value={})
+    def test_get_matching_missing_key(self, _get_unverified_header_mock: MagicMock) -> None:
+        with pytest.raises(Unauthorized):
+            get_matching_jwk("x")
+
+
+class TestDecodeJWT:
+    @patch("lbz.jwt_utils.get_matching_jwk", return_value={})
+    def test_missing_public_keys(self, get_matching_jwk_mock: MagicMock) -> None:
```

Comment:
Based on the code, if public keys are missing, we should get the `RuntimeError` 🤔  Can you verify that?